sortinghat/core/api.py

```diff
@@ -714,42 +714,84 @@ def withdraw(uuid, organization, from_date=None, to_date=None):
-    assigned to `to_uuid`. In addition, `from_uuid` will be removed
+    Use this function to join a list of `from_uuid` unique identities into
+    `to_uuid`. Identities and enrollments related to each `from_uuid` will be
+    assigned to `to_uuid`. In addition, each `from_uuid` will be removed
     from the registry. Duplicated enrollments will be also removed from
     the registry while overlapped enrollments will be merged.
 
-    This function also merges two profiles. When a field on `to_uuid`
+    This function also merges two or more profiles. When a field on `to_uuid`
     profile is `None` or empty, it will be updated with the value on the
-    profile of `from_uuid`. If any of the two unique identities was set
+    profile of each `from_uuid`. If any of the unique identities was set
     as a bot, the new profile will also be set as a bot.
 
-    When `from_uuid` and `to_uuid` are equal, or any of this two fields
-    are `None` or empty, it raises an `InvalidValueError`.
+    When any of the `from_uuid` and `to_uuid` are equal, or any of this two
+    fields are `None` or empty, it raises an `InvalidValueError`.
 
-    The function raises a `NotFoundError` exception when either `from_uuid`
+    The function raises a `NotFoundError` exception when either any `from_uuid`
     or `to_uuid` do not exist in the registry.
 
-    :param from_uuid: identifier of the unique identity set to merge
-    :param to_uuid: identifier of the unique identity where `from_uuid`
+    :param from_uuids: List of identifiers of the unique identities set to merge
+    :param to_uuid: identifier of the unique identity where `from_uuids`
         will be merged
 
     :returns: a UniqueIdentity object as the result of the merged identities
 
-    :raises NotFoundError: raised when either `from_uuid` or `to_uuid`
-        do not exist in the registry
+    :raises NotFoundError: raised when either one of the `from_uuids` or
+        `to_uuid` do not exist in the registry
     """
-    def _merge_enrollments(trxl, from_uid, to_uid):
-        """Merge enrollments from two `UniqueIdentity` objects"""
-        # Get current enrollments from both uidentities
-        enrollments_db = from_uid.enrollments.all() | to_uid.enrollments.all()
-        enrollments = {}
 
+    def _find_unique_identities(from_uuids, to_uuid):
+        """Find the unique identities to be merged from their uuids"""
+
+        unique_identities = []
+        for from_uuid in from_uuids:
+            # Check whether input values are valid
+            if from_uuid is None:
+                raise InvalidValueError(msg="'from_uuid' cannot be None")
+            if from_uuid == '':
+                raise InvalidValueError(msg="'from_uuid' cannot be an empty string")
+            if from_uuid == to_uuid:
+                raise InvalidValueError(msg="'from_uuid' and 'to_uuid' cannot be equal")
+
+            try:
+                unique_identities.append(find_unique_identity(from_uuid))
```

Comment:
Would you mind to store the uidentity first in a variable and then append that variable to the list? I think that's more clear.